From 35f331ecd955345def561b1b8fd85bee07df7161 Mon Sep 17 00:00:00 2001
From: Mitali Vagadiya <vmnagajibhai@habana.ai>
Date: Fri, 06 Jun 2025 12:21:33 +0300
Subject: [PATCH] [JSW-705]  hls2-bmc: don't search gpu specs

hls2 and ami post script is serching and copying common headers
and dev specific headers in order to build habanalabs utils
such as fw-loader.
For GPU project some common headers are duplicated in
order to start from a clean slate, and avoid dragging Gaudi defines.
This duplication creates an issue for the script searching for headers.
Since JGS devices will never work with HLS2 as well as AMI,
build should not look for GPU headers.
This patch adds a directive to avoid serching specs/gpu folder
for headers.

Change-Id: I4b44d9b341f7d9e197e1260efcbd2201f768f7e3
---

diff --git a/board/habanalabs/common/post-ami-build.sh b/board/habanalabs/common/post-ami-build.sh
index 9cf7268..f0a7eff 100755
--- a/board/habanalabs/common/post-ami-build.sh
+++ b/board/habanalabs/common/post-ami-build.sh
@@ -188,7 +188,7 @@
 		file_found=0
 		for specs_dir in ${SPECS_SEARCH_DIR[@]}
 		do
-		file_abs_path=$(find "${specs_dir}/${project}" -not \( -path ${specs_dir}/${project}/asic_reg_structs -prune -o -path ${specs_dir}/fs -prune \) -name ${file})
+		file_abs_path=$(find "${specs_dir}/${project}" -not \( -path ${specs_dir}/${project}/asic_reg_structs -prune -o -path ${specs_dir}/fs -prune -o -path "${specs_dir}/gpu" -prune \) -name ${file})
 		if [ ! -z "$file_abs_path" ]
 		then
 			file_found=1
diff --git a/package/hls2-bmc/deploy-shared-sources.sh b/package/hls2-bmc/deploy-shared-sources.sh
index a79671f..5ca3e7c 100755
--- a/package/hls2-bmc/deploy-shared-sources.sh
+++ b/package/hls2-bmc/deploy-shared-sources.sh
@@ -145,7 +145,7 @@
 		for specs_dir in ${SPECS_SEARCH_DIR[@]}
 		do
 		file_abs_path=$(find "${specs_dir}/${project}" \
-			-not \( -path ${specs_dir}/${project}/asic_reg_structs -prune -o -path ${specs_dir}/fs -prune \) -name ${file})
+			-not \( -path ${specs_dir}/${project}/asic_reg_structs -prune -o -path ${specs_dir}/fs -prune -o -path "${specs_dir}/gpu" -prune \) -name ${file})
 		if [ ! -z "$file_abs_path" ]
 		then
 			file_found=1
